### PR TITLE
pythonPackages: add pyhepmc, graphviz; bump rootpy

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -15341,6 +15402,25 @@ in {
     };
   };
 
+  graphviz = buildPythonPackage rec {
+    name = "graphviz-${version}";
+    version = "0.5.2";
+    disable = isPy3k;
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/01/98/8dec899491e4ac01a6fc8269e3b6dffd35421321b6858e21672489678fa8/graphviz-0.5.2.zip";
```
Comment (lsix) [inline pkgs/top-level/python-packages.nix]: You could replace this with:

```
url = "mirror://pypi/g/graphviz/${name}.zip";
```
This makes updating the derivation easier singe changing the `version` attribute is enough to update the url as well.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -15341,6 +15402,25 @@ in {
     };
   };
 
+  graphviz = buildPythonPackage rec {
+    name = "graphviz-${version}";
+    version = "0.5.2";
+    disable = isPy3k;
```
Comment (lsix) [inline pkgs/top-level/python-packages.nix]: According to the [project page](https://github.com/xflr6/graphviz): “This package runs under Python 2.6, 2.7, and 3.3+”.

The [setup.py](https://github.com/xflr6/graphviz/blob/289f0ea41c9d1e1ff1b51225ba5a8820dc019c89/setup.py) has:

```python
        'Programming Language :: Python :: 2',
        'Programming Language :: Python :: 2.6',
        'Programming Language :: Python :: 2.7',
        'Programming Language :: Python :: 3',
        'Programming Language :: Python :: 3.3',
        'Programming Language :: Python :: 3.4',
        'Programming Language :: Python :: 3.5',
```

as classifiers.

Any reason to disable this derivation for python3?

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -22807,12 +22887,12 @@ in {
...
 
   rootpy = buildPythonPackage rec {
-    version = "0.8.3";
+    version = "0.9.0";
     name = "rootpy-${version}";
 
     src = pkgs.fetchurl {
-      url = "https://pypi.python.org/packages/d5/40/feddb2c9d1cadfe05d1d9aea1a71be093dc700879c9f6af40a10b1330f34/rootpy-0.8.3.tar.gz";
-      sha256 = "14q9bhs2a53598571x8yikj68x2iyl6090wbvdrpbwr238799b0z";
+      url = "https://pypi.python.org/packages/c0/a4/2672dad1920c23375fd8155c8fd7940b7e36fdd3cd018a886420dc6c76d0/rootpy-0.9.0.tar.gz";
```
Comment (lsix) [inline pkgs/top-level/python-packages.nix]: Here you could also use `mirror://pypi/r/rootpy/${name}.tar.gz` as a simplification.

**Comment by lsix** [comment]
Hi @veprbl,

Here is the error from travis-ci (which is the same I got locally if A try to build this PR):

```
trying https://bitbucket.org/lukas_heinrich/pyhepmc/commits/325f89b7b72922e9add9ca9dd0f7ca4a6c83bf00/raw

output path ‘/nix/store/gfp9la8vwsbihdc5csdkm9hpxdv1hq3y-raw’ has sha256 hash ‘0rcx9hksd9aav0mycpycbwcgc7sy1hrkjyknn1y8ljspy31fg7id’ when ‘01msflk0zs7yvijqip1swykfw35g7f30628bsh9hhnk1bfyanf8b’ was expected

cannot build derivation ‘/nix/store/9q3c25hshkpia6cxjw6izw2i8l6assyq-python2.7-pyhepmc-0.5.0.drv’: 1 dependencies couldn't be built
```

Apparently, the hash of the patch you fetch for pyhepmc do not match.

This is quite strange because I can run:

```
$ nix-prefetch-url https://bitbucket.org/lukas_heinrich/pyhepmc/commits/325f89b7b72922e9add9ca9dd0f7ca4a6c83bf00/raw
downloading ‘https://bitbucket.org/lukas_heinrich/pyhepmc/commits/325f89b7b72922e9add9ca9dd0f7ca4a6c83bf00/raw’... [0/0 KiB, 0.0 KiB/s]
path is ‘/nix/store/gfp9la8vwsbihdc5csdkm9hpxdv1hq3y-raw’
01msflk0zs7yvijqip1swykfw35g7f30628bsh9hhnk1bfyanf8b
```

Did you have any trouble when building your derivation?

**Comment by veprbl** [comment]
I can reproduce exactly your result. I'm not sure what is going on, probably bitbucket gets triggered by some different http headers that fetchpatch could send. I will just put the patch itself to the repo. Hope that's alright.

**Comment by lsix** [comment]
Builds fine, merged.

Thanks.

**Comment by veprbl** [comment]
Thank you for review!

**Pull request merged**